pymlx/evaluation.py

```diff
@@ -69,20 +69,26 @@ def confusion_matrix(truths, predictions, sampling_rate=1):
 
 
 def diagnose(model, test_df, truths, cols=None, predictions=None, good_to_bad=False, pretty=True):
-    assert isinstance(model, BinaryClassifier)
-    test_features = model.featurizer.transform(test_df)
+    # assert isinstance(model, BinaryClassifier)
+
+    test_features = model.featurizer.transform(test_df, return_dataframe=False)
     if predictions is None:
-        predictions = model.bulk_predict(test_features)
+        predictions, _ = model.bulk_predict(test_df)
 
-    df = DataFrame(test_features.todense(),
+    df = DataFrame(test_features.tolist(),
                    columns=model.featurizer.out_feature_names,
                    index=test_df.index)
     status_cols = ['Labeled', 'Predicted Probability', 'diff']
+
     for i, status in enumerate([truths, predictions,
                                 [abs(truths[i] - p) for i, p in enumerate(predictions)]]):
         df[status_cols[i]] = status
+
+    if cols is None:
+        cols = [item for item in df.columns if item not in status_cols]
+
     df = df[status_cols + list(cols)].sort_values('diff', ascending=good_to_bad)
     df.drop('diff', axis=1, inplace=True)
     if pretty:
-        df.columns = [name.title().replace('_', ' ') for name in df.columns]
+        df.columns = [name.title().replace('_', ' ').replace('=', ': ') for name in df.columns]
```

Comment:
Why `.replace('=', ': ')`? If we prefer `:`, why don't we change the naming in the handlers?